perun-rpc/src/main/java/cz/metacentrum/perun/rpc/Api.java

```diff
@@ -225,6 +225,8 @@ private static PerunPrincipal setupPerunPrincipal(HttpServletRequest req, Deseri
 		int extSourceLoa;
 		Map<String, String> additionalInformations = new HashMap<>();
 
+		String referer = req.getHeader("Referer");
```

Comment:
I'm not really sure what this logic actually suppose to do, but HTTP request headers are set by client. Hence I can forge a request using cURL with specific `Referer` value, even it if originates from my script. If this suppose to somehow override the problems with MFA, I would be cautious.

Reply:
According to the referer header, this code removes roles from a session that are not defined as allowed in the config. It does not skip MFA but only limits user's rights. So even if a user fakes the request, there should be no security risk.